### ISSUE TITLE
Can BERT really handle misspelled words?

### ISSUE THREAD

**Issue by chikubee** [google-research/bert#812]
It is said that Tokenizers like sentencepiece and wordpiece can handle misspelled words., 
How does BERT really handle them?

I fine-tuned pretrained BERT base uncased model for Sentiment Analysis Task,
The prediction completely changes due to a small spelling mistake.

What could be the reason for the masked tokens to get higher attention.

Thanks in advance.
<img width="566" alt="Screenshot 2019-08-19 at 6 04 23 PM" src="https://user-images.githubusercontent.com/25073753/63266283-3ffa2680-c2ad-11e9-9bea-349faf32f482.png">
<img width="587" alt="Screenshot 2019-08-19 at 6 04 50 PM" src="https://user-images.githubusercontent.com/25073753/63266284-3f619000-c2ad-11e9-9838-89fc8afd1833.png">

**Comment by LifeIsStrange** [comment]
I don't know if BERT can handle misspelled words.
It would be a significant improvement if BERT optionally integrated a lexical normalizer, because many real world writings are partially incorrects but trivially correctable by a human reader.

https://github.com/sebastianruder/NLP-progress/blob/master/english/lexical_normalization.md

**Comment by chikubee** [comment]
@LifeIsStrange I was wrong to really call it misspelled, this is kind of an abbreviation "cud" vs "could". 
What I fail to understand is how these masked tokens get so much attention?
<img width="581" alt="Screenshot 2019-08-20 at 11 31 42 AM" src="https://user-images.githubusercontent.com/25073753/63321365-23f29580-c33e-11e9-948f-9b27891fbd4d.png">
Another example where we can see ##i being more attentive than other tokens.

Should the other words in the sentence not be the driving factor in determining the sentiment 
than giving preference to the words that are not even there in the dictionary?

**Comment by LifeIsStrange** [comment]
@chikubee I'm sorry I am not knowledgeable enough to answer your interrogations!
I've never used BERT so.. I've not even yet understood what are masked tokens :m
Maybe you can ping some devs with @ so you get an answer faster? 
Or try to ask on stackoverflow.com! 

I can only indirectly help you:
As said precedently lexical normalization can unabbreviate your problematic words. 

Otherwise you can try to use
https://github.com/zihangdai/xlnet
It is the new state of the art, beyond BERT and *maybe* it does not suffer this issue.

**Comment by shashankmc** [comment]
> I don't know if BERT can handle misspelled words.
> It would be a significant improvement if BERT optionally integrated a lexical normalizer, because many real world writings are partially incorrects but trivially correctable by a human reader.

Absolutely, it would be really good if BERT had an option for lexical normaliser. I'm currently facing that issue when using BERT to analyse transcripts from the 80s. So many misspelled words!

**Comment by LifeIsStrange** [comment]
@shashankmc you would also benefit from
 https://github.com/sebastianruder/NLP-progress/blob/master/english/grammatical_error_correction.md
BERT needs to provide grammatical error correction but ironically the best error correcters are fine tuned BERT :')

**Comment by shashankmc** [comment]
@LifeIsStrange Thank you! And ahah, I know what you mean. I had the repo starred and can't believe I haven't noticed this before. 🙄
